packages/semver/src/executors/version/version.ts

```diff
@@ -83,16 +80,22 @@ export function versionWorkspace({
           )
         )
       ),
-      concatMap((packageFiles) =>
-        addToStage({
-          paths: packageFiles.filter(
-            (packageFile) => packageFile !== null
-          ) as string[],
-          dryRun,
-        })
+      map(
+        (packageFiles) =>
+          packageFiles.filter((packageFile) => packageFile !== null) as string[]
       )
     ),
   ]).pipe(
+    concatMap((pathsToStage) => {
+      const pathsSet = new Set<string>();
+      pathsToStage.forEach((paths) =>
+        paths.forEach((path) => pathsSet.add(path))
+      );
```

Comment:
It shouldn't matter, I had it originally since I wasn't looking that hard at where the strings came from and didn't know if they could be the same or not and didn't want duplicates. Doesn't matter to me either way, so if you think they're fine to just all flatten/concat into a single array then can do that too.

Reply:
```
    concatMap((pathsToStage) => {
      const paths = pathsToStage.reduce((agg, paths) => agg.concat(paths), []);
      return addToStage({
        paths,
        dryRun,
      });
    }),
```

Reply:
`pathsToStage` is an array of array now? maybe we can just flatten it: 

```ts
   concatMap((paths) => {
      return addToStage({
        paths: paths.flat(),
        dryRun,
      });
    }),
```

Reply:
Yea, it's a tuple from the results of the two parallel calls that both return an array of paths.

On another note, I'm not sure how I've never seen flat(), but that looks better. 😄

Reply:
It's an es2019 feature, not so used.